server/src/main/java/org/apache/uniffle/server/ShuffleServer.java

```diff
@@ -78,6 +82,10 @@ public class ShuffleServer {
   private AtomicBoolean isHealthy = new AtomicBoolean(true);
   private GRPCMetrics grpcMetrics;
   private MetricReporter metricReporter;
+  private Thread decommissionThread;
+  private ServerStatus serverStatus = ServerStatus.NORMAL_STATUS;
+  private Object statusLock = new Object();
+  private boolean running;
```

Comment:
Need volatile?

```suggestion
  private volatile boolean running;
```

Reply:
Feels unnecessary, It just for UT now. But it is OK for me too.